### ISSUE TITLE
[sklearn#34404] Backport workflow-hardening fix (excessive-permissions) to 4 release branches

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#327]
**Mirrored from [scikit-learn#34404](https://github.com/scikit-learn/scikit-learn/issues/34404)**

Original title: Backport workflow-hardening fix (excessive-permissions) to 4 release branches
Original labels: 
Created: 2026-06-29T02:57:21Z

---

### Summary
The default branch already hardened `.github/workflows/cuda-ci.yml`, `.github/workflows/update-lock-files.yml`, `.github/workflows/wheels.yml` against the issue(s) below, but **4** release branches still carry it. This proposes the same, minimal, scanner-verified fix for each.

### What's flagged (by [zizmor](https://github.com/woodruffw/zizmor))
- `excessive-permissions` — workflow/job granted broader `permissions` than needed

Already resolved on the default branch in:
- https://github.com/scikit-learn/scikit-learn/commit/0ae1c431bcd4800c7d7902d9d56c834f3982e97f
- https://github.com/scikit-learn/scikit-learn/commit/94e517ecad92e28c870555120bdac76c9baba0dc
- https://github.com/scikit-learn/scikit-learn/commit/cfcb06a40ef2b668f4871ab5e1c20e18b1798bc8

### Affected release branches (4)
- **`1.4.X`** (still present as of HEAD `e8dd30c8`)
- **`1.6.X`** (still present as of HEAD `7bd51a6e`)
- **`1.5.X`** (still present as of HEAD `7d7289f9`)
- **`1.3.X`** (still present as of HEAD `4b18f559`)

### Suggested per-branch patches
Each diff below was checked locally with **zizmor** and **actionlint**: the flagged finding(s) are cleared on the affected construct and no new lint or security findings are introduced. (Whitespace is normalized; only security-relevant lines change.)

<details>
<summary><code>1.4.X</code> &mdash; excessive-permissions</summary>

File `.github/workflows/update-lock-files.yml`; suggested edits:
  - + permissions.contents = 'read'

```diff
--- a/.github/workflows/update-lock-files.yml
+++ b/.github/workflows/update-lock-files.yml
@@ -69,3 +69,5 @@
           echo "- **URL** ${{ steps.cpr.outputs.pull-request-url }}" >> ${GITHUB_STEP_SUMMARY}
           echo "- **Operation** [${{ steps.cpr.outputs.pull-request-operation }}]" >> ${GITHUB_STEP_SUMMARY}
           echo "- **SHA** ${{ steps.cpr.outputs.pull-request-head-sha }}" >> ${GITHUB_STEP_SUMMARY}
+permissions:
+  contents: read
```
</details>
<details>
<summary><code>1.6.X</code> &mdash; excessive-permissions</summary>

File `.github/workflows/cuda-ci.yml`; suggested edits:
  - + permissions.contents = 'read'

```diff
--- a/.github/workflows/cuda-ci.yml
+++ b/.github/workflows/cuda-ci.yml
@@ -74,3 +74,5 @@
           SCIPY_ARRAY_API=1 pytest --pyargs sklearn -k 'array_api' -v
         # Run in /home/runner to not load sklearn from the checkout repo
         working-directory: /home/runner
+permissions:
+  contents: read
```
</details>
<details>
<summary><code>1.5.X</code> &mdash; excessive-permissions</summary>

File `.github/workflows/update-lock-files.yml`; suggested edits:
  - + permissions.contents = 'read'

```diff
--- a/.github/workflows/update-lock-files.yml
+++ b/.github/workflows/update-lock-files.yml
@@ -69,3 +69,5 @@
           echo "- **URL** ${{ steps.cpr.outputs.pull-request-url }}" >> ${GITHUB_STEP_SUMMARY}
           echo "- **Operation** [${{ steps.cpr.outputs.pull-request-operation }}]" >> ${GITHUB_STEP_SUMMARY}
           echo "- **SHA** ${{ steps.c

...(truncated)

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/96 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
